packages/compiler/src/core/checker.ts

```diff
@@ -2928,12 +2928,30 @@ export function createChecker(program: Program, resolver: NameResolver): Checker
       }
     } else if (identifier.parent && identifier.parent.kind === SyntaxKind.MemberExpression) {
       let base = resolver.getNodeLinks(identifier.parent.base).resolvedSymbol;
-      if (base) {
-        if (base.flags & SymbolFlags.Alias) {
-          base = getAliasedSymbol(base, undefined);
+
+      if (identifier.parent.selector === "::") {
+        if (base?.node === undefined && base?.declarations) {
+          // Process meta properties separately, such as `::parameters`, `::returnType`
+          for (const nodeModels of Object.values(base?.declarations)) {
```

Comment:
i think you can use declaration[0] directly, (make sure to check the length first)